server/src/domain/system-config/system-config.service.ts

```diff
@@ -53,6 +79,70 @@ export class SystemConfigService {
     this.core.addValidator(validator);
   }
 
+  async handleImmichLatestVersionAvailable() {
+    const proxy =
+      this.proxyHost && this.proxyPort && this.proxyProtocol
+        ? {
+            protocol: this.proxyProtocol,
+            port: this.proxyPort,
+            host: this.proxyHost,
+          }
+        : false;
+
+    if (this.disableCheckLatestVersion) {
+      return true;
+    }
+    try {
+      const { data } = await axios.get<GithubRelease>(
+        'https://api.github.com/repos/immich-app/immich/releases/latest',
+        {
+          headers: {
+            Accept: 'application/vnd.github.v3+json',
+          },
+          proxy,
+        },
+      );
+
+      const [, major, minor, patch] = data.tag_name.match(/v(\d+)\.(\d+)\.(\d+)/) || [];
+
+      const GithubVersion = {
+        major: parseInt(major, 10),
+        minor: parseInt(minor, 10),
+        patch: parseInt(patch, 10),
+      };
+
+      const config = await this.core.getConfig();
+      if (
+        GithubVersion.major > serverVersion.major ||
+        GithubVersion.minor > serverVersion.minor ||
+        GithubVersion.patch > serverVersion.patch
+      ) {
```

Comment:
Indeed ! Maybe if only one is different ?

Reply:
There should be libraries out there that implement semver comparison. The logic isn't particularly complex, but it might be cleaner to just pull one of those in?

Reply:
You can change the `serverVersion` in `domain.constant.ts` to be a class like so:
```
export class ServerVersion implements IServerVersion {
  major: number;
  minor: number;
  patch: number;

  constructor(major: string | number, minor: string | number, patch: string | number) {
    this.major = Number(major);
    this.minor = Number(minor);
    this.patch = Number(patch);
  }

  toString() {
    return `${this.major}.${this.minor}.${this.patch}`;
  };

  valueOf() {
    return this.major * 100000 + this.minor * 1000 + this.patch;
  }
}
```

If you use this class for the GH version as well, then you can compare in two ways.

1. `githubVersion > serverVersion`
    - Very elegant, but it assumes the minor and patch versions will be less than 100)
2. `githubVersion.toString().localeCompare(serverVersion.toString(), undefined, {numeric: True}) === 1`
    - Less readable, but the `numeric` property will ensure this works correctly in all cases (see [here](https://stackoverflow.com/a/55466325))

Reply:
Just use this and call it a day. Almost 300k weekly downloads. It's the implementation npm uses.

Reply:
If Immich keeps the current semantic versionning, it's probably better to avoid an extra dependency ? I like a lot the second option proposed by @mertalev